File: src/app/models/Transaction.js

// Define collection and schema for Transaction
export interface Transaction {
    bookingDate:
	type : Date
    valueDate:
	type : Date
    amount:
	type : Money
    description:
	type : string
    Account:
	type : Schema.Types.ObjectId
    ExternalCounterparty:
	type : Schema.Types.ObjectId
    PaymentCard:
	type : Schema.Types.ObjectId
    FundsTransfer:
	type : Schema.Types.ObjectId
    FxTrade:
	type : Schema.Types.ObjectId
    Dispute:
	type : Schema.Types.ObjectId
    Direction:
 	type : String
    TransactionType:
 	type : String
    Status:
 	type : String
    Channel:
 	type : String
#
    collection: 'transactions'
}

File: bankingOnGolang/internal/model/Transaction.go

package model

import (
    "time"
)

//==============================================================
// Transaction Declaration
//==============================================================
type Transaction struct {
    BaseModel
     BookingDate            time.Time
    ValueDate            time.Time
    Amount            Money
    Description            string
    AccountId         *uint
    Account           *Account `gorm:"foreignKey:AccountId"`
    ExternalCounterpartyId         *uint
    ExternalCounterparty           *ExternalAccount `gorm:"foreignKey:ExternalCounterpartyId"`
    PaymentCardId         *uint
    PaymentCard           *PaymentCard `gorm:"foreignKey:PaymentCardId"`
    FundsTransferId         *uint
    FundsTransfer           *FundsTransfer `gorm:"foreignKey:FundsTransferId"`
    FxTradeId         *uint
    FxTrade           *FXTrade `gorm:"foreignKey:FxTradeId"`
    DisputeId         *uint
    Dispute           *Dispute `gorm:"foreignKey:DisputeId"`
    Direction            TransactionDirection
    TransactionType            TransactionType
    Status            TransactionStatus
    Channel            ChannelType

// parent associations as their child

}

